app/controllers/serviceLearning/routes.py

```diff
@@ -87,9 +87,17 @@ def slcEditProposal(courseID):
 def slcCreateCourse():
     """will give a new course ID so that it can redirect to an edit page"""
     course = createCourse(g.current_user)
-
     return redirect(url_for('serviceLearning.slcEditProposal', courseID = course.id))
 
+@serviceLearning_bp.route('/serviceLearning/canceledProposal', methods=['POST'])
+def slcCancelProposal():
+    courseID = request.form.get('courseID')
+    course = Course.get_by_id(courseID)
+    if not course.courseName and not course.courseAbbreviation:
+        CourseQuestion.delete().where(CourseQuestion.course_id == courseID).execute()
```

Comment:
Once we have gotten a Course object, we should use it everywhere, rather than reverting back to a value passed in to the function (`courseID`). And with peewee, you don't need to compare id to id. It's smart enough to compare the objects themselves. So, `CourseQuestion.course == course` would be the preferred conditional here.

Reply:
Changed it to CourseQuestion.course == course